packages/spring/test/index.test.ts

```diff
@@ -0,0 +1,164 @@
+import { describe, expect, it } from "vitest";
+import { createRoot, createSignal } from "solid-js";
+import { createDerivedSpring, createSpring } from "../src/index.js";
+
+// ===========================================================================
+// Tests
+// ===========================================================================
+
+describe("createSpring", () => {
+  it("returns values", () => {
+    const { value, setValue, dispose } = createRoot(dispose => {
+      const [value, setValue] = createSpring({ progress: 0 });
+      expect(value().progress, "initial value should be { progress: 0 }").toBe(0);
+
+      return { value, setValue, dispose };
+    });
+
+    setValue({ progress: 100 });
+    dispose();
+  });
+
+  it("updates toward target", async () => {
+    let dispose!: () => void;
+    let setSpringed!: (
+      newValue: number,
+      opts?: { hard?: boolean; soft?: boolean },
+    ) => Promise<void>;
+    let springed!: () => number;
+
+    createRoot(d => {
+      dispose = d;
+      [springed, setSpringed] = createSpring(0);
+    });
```

Comment:
You can just return from `createRoot`
```suggestion
	const {dispose, springed, setSpringed} = createRoot(dispose => {
		const [springed, setSpringed] = createSpring(0)
		return {dispose, springed, setSpringed}
	})
```